Added more information to CONTRIBUTING.md (#14)
courselab/snaskii21: doc/CONTRIBUTING.md

beta, release candidates etc.) reside in branch _prerelease_. Late fixes in
past stabe releases for maintenance, if any, are found in _legacy_ branches.

It's important that every pull request be linked to an issue. You can pick
one that already exists or create one and assign yourself to it.

First step to make a contribution is creating a branch. The branch created must
follow the naming convention bellow, where `<issue-number>` is the number of the
issue that the pull request will solve (e.g. #42).

```

> Branches names should be one of the four options bellow:

Reviewer: We could provide a brief description about each branch. For example, when should the contributor use coldfix instead of hotfix? This would also make the CONTRIBUTING.md doc more self-contained. What do you think?

Author: I actually tried to do this but failed to describe the terms in a good way and just gave up (maybe it's my lack of experience/knowledge on these different terms), but I think it's a good idea. Just need some help with that.

Reviewer 2: `feature/<issue-number>` -> Implementation of a new feature. Maybe now with[ this response](https://github.com/courselab/snaskii21/issues/10#issuecomment-965285862) from @monacofj we could consider the missing "p" implementation as a feature, not a bug.

`hotfix/<issue-number>` -> Fixes something that is broken in production. This needs to be merged asap, it usually means clients are complaining and company/organization is losing money or reputation.

`wip/<issue-number>` -> Work in progress, usually tagged with "DO NOT MERGE" as well. This is for helping with early feedback on some development that will end up being a large PR.

`coldfix/<issue-number>` -> Not too familiar with this one, but my guess is that it fixes an issue that has gone cold (inactive for a long time).

Of course all fo these can be better elaborated, just wanted to give a general sence of it. It shouldn't be hard to find these definitions online, though.

Reviewer: AFAIK coldfix usually refers to planned fixes, which are announced before hand, so that users know when and how long it will take. The others descriptions seem good to give an overall idea of each branch, thanks!

>  Maybe now with this response from @monacofj we could consider the missing "p" implementation as a feature, not a bug.

@Float07 flagged it as a feature and as critical (high priority issue), not as a bug. So I think that his flags are coherent with the answer from @monacofj in #10 .

Author: Thanks a lot for the help guys! I added these info to the document. 
I also added that a `feature` branch covers updates to existing features and updates to documentation. This information is correct, right?

Just let me know if there's something wrong in this addition 😉

Reviewer: > I also added that a `feature` branch covers updates to existing features and updates to documentation. This information is correct, right?

I believe so. Thanks for the updates! :)

beta, release candidates etc.) reside in branch _prerelease_. Late fixes in
past stabe releases for maintenance, if any, are found in _legacy_ branches.

> It's important that every pull request be linked to an issue. You can pick

Reviewer 2: ```suggestion
It's important that every pull request fixes at least one issue. You can pick
```

> an issue that already exists or create one yourself. After an issue has been

Reviewer 2: ```suggestion
an issue that already exists or you can create one yourself, if you can't find one that addresses your problem. After an issue has been
```

> selected, assign yourself to it to notify other collaborators that you are

Reviewer 2: ```suggestion
selected, assign yourself so other collaborators can be aware that you are
```

Author: Hey thanks for all your suggestions, I think they help to make the text clearer 😄  . 
Just be aware that you can make one single suggestion for multiple successive lines, making it easier to apply it. Take a look in step 4 of [this document](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request).
